From ea8ab0c9205341175ff3808885bf5431d5bbd32b Mon Sep 17 00:00:00 2001
From: Carlo Landmeter <clandmeter@gmail.com>
Date: Mon, 28 Sep 2015 13:08:16 +0200
Subject: [PATCH] abuild: check if depends_dev is set unnecessarily

depends_dev are dependencies of the of the developement subpackage
it does not need to be set when there is no developement subpackage
---
 abuild.in | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/abuild.in b/abuild.in
index fd6fb53..995768a 100644
--- a/abuild.in
+++ b/abuild.in
@@ -212,6 +212,8 @@ default_sanitycheck() {
 
 	check_maintainer || die "Provide a valid RFC822 maintainer address"
 
+	check_depends_dev || warning "depends_dev found but no development subpackage found"
+
 	makedepends_has 'g++' && warning "g++ should not be in makedepends"
 	return 0
 }
@@ -729,6 +731,8 @@ get_maintainer() {
 	fi
 }
 
+
+
 check_maintainer() {
 	get_maintainer
 	if [ -z "$maintainer" ]; then
@@ -742,6 +746,18 @@ check_maintainer() {
 	fi
 }
 
+check_depends_dev() {
+	if [ -n "$depends_dev" ]; then
+		local pkgname
+		for i in $pkgname $subpackages; do
+			case "$pkgname" in
+				*-dev)  return 0 ;;
+			esac
+		done
+		return 1
+	fi
+}
+
 prepare_metafiles() {
 	getpkgver || return 1
 	local name=${subpkgname:-$pkgname}
-- 
2.5.3

